src/main/java/List.java

```diff
@@ -0,0 +1,165 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class List {
+    ArrayList<Task> tasks =new ArrayList<>();
+    private static int size;
+
+    public List() {
+        this.size = 0;
+    }
+
+    public void printList() {
+        for (int i=0;i<tasks.size();i++) {
+            System.out.println(i+1+". " +tasks.get(i).toString());
+        }
+        System.out.println("");
+    }
+
+
+    public void addDeadline (String description, String by) {
+        tasks.add(new Deadline(description,by));
+        System.out.println("Got it. I've added this task:");
+        System.out.println(tasks.get(this.size).toString());
+        this.size++;
+        System.out.println("Now you have " + this.size + " tasks in the list.\n");
+    }
+
+    public void addOldDeadline (String description, String by, Boolean isDone) {
+        tasks.add(new Deadline(description,by,isDone));
+        this.size++;
+    }
+
+
+    public void addEvent(String description, String at) {
+        tasks.add(new Event(description, at));
+        System.out.println("Got it. I've added this task:");
+        System.out.println(tasks.get(this.size).toString());
+        this.size++;
```

Comment:
You do not need to use `this` keyword if it is not ambiguous (shadowed by a method or constructor parameter).